hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/io/TestHoodieTimelineArchiver.java

```diff
@@ -502,38 +493,26 @@ public void testSavepointWithArchival(boolean archiveBeyondSavepoint) throws Exc
         getActiveCommitInstants(Arrays.asList("00000008", "00000009")), commitsAfterArchival);
   }
 
-  @ParameterizedTest
-  @ValueSource(booleans = {true, false})
-  public void testMergeSmallArchiveFilesRecoverFromBuildPlanFailed(boolean enableArchiveMerge) throws Exception {
-    HoodieWriteConfig writeConfig = initTestTableAndGetWriteConfig(true, 4, 5, 2, enableArchiveMerge, 3, 209715200);
+  @Test
+  public void testCompactionWithCorruptVersionFile() throws Exception {
+    HoodieWriteConfig writeConfig = initTestTableAndGetWriteConfig(true, 4, 5, 2, 3);
 
     // do ingestion and trigger archive actions here.
     for (int i = 1; i < 10; i++) {
       testTable.doWriteOperation("0000000" + i, WriteOperationType.UPSERT, i == 1 ? Arrays.asList("p1", "p2") : Collections.emptyList(), Arrays.asList("p1", "p2"), 2);
       archiveAndGetCommitsList(writeConfig);
     }
 
-    // build a merge small archive plan with dummy content
```

Comment:
I see. you are replacing the existing small archived file merging impl with the new LSM based one